spec/create.spec.js

```diff
@@ -57,88 +56,138 @@ describe('cordova create checks for valid-identifier', function () {
 
 describe('create end-to-end', function () {
 
-    function checkProject () {
+    function checkProjectCommonArtifacts () {
         // Check if top level dirs exist.
         var dirs = ['hooks', 'platforms', 'plugins', 'www'];
         dirs.forEach(function (d) {
             expect(path.join(project, d)).toExist();
         });
 
+        // Check that README.md exists inside of hooks
         expect(path.join(project, 'hooks', 'README.md')).toExist();
 
-        // Check if www files exist.
+        // Check that index.html exists inside of www
         expect(path.join(project, 'www', 'index.html')).toExist();
 
-        // Check that config.xml was updated.
-        var configXml = new ConfigParser(path.join(project, 'config.xml'));
-        expect(configXml.packageName()).toEqual(appId);
-
-        // TODO (kamrik): check somehow that we got the right config.xml from the fixture and not some place else.
-        // expect(configXml.name()).toEqual('TestBase');
-    }
+        // Check if config.xml exists.
+        expect(path.join(project, 'config.xml')).toExist();
 
-    function checkConfigXml () {
-        // Check if top level dirs exist.
-        var dirs = ['hooks', 'platforms', 'plugins', 'www'];
-        dirs.forEach(function (d) {
-            expect(path.join(project, d)).toExist();
-        });
-        expect(path.join(project, 'hooks', 'README.md')).toExist();
-
-        // index.js and template subdir folder should not exist (inner files should be copied to the project folder)
+        // index.html, index.js and template subdir folder
+        // should not exist in top level
+        // (inner files should be copied to the project top level folder)
+        expect(path.join(project, 'index.html')).not.toExist();
         expect(path.join(project, 'index.js')).not.toExist();
         expect(path.join(project, 'template')).not.toExist();
 
-        // Check if www files exist.
-        expect(path.join(project, 'www', 'index.html')).toExist();
+        // Check that .gitignore does not exist inside of www
+        expect(path.join(project, 'www', '.gitignore')).not.toExist();
+
+        // Check that .npmignore does not exist inside of www
+        expect(path.join(project, 'www', '.npmignore')).not.toExist();
```

Comment:
I would remove all mentions of `.npmignore` from the specs. I think we should really treat it like any other file. To me, these explicit checks convey that we are handling these files specially.

Reply:
I would personally favor keeping this check to keep the related behavior as clear as possible.

Reply:
But our intended behavior is `Treat it like any other file`. So if a template wanted to define a `.npmignore`, it could. IMO, the only tests related to `.npmignore` (if any) should be to check that it is preserved in the project if we have on in the template (or a `.gitignore` that is being transformed to one).

That being said, this is no show stopper for me.